schema/namespace.go

```diff
@@ -0,0 +1,62 @@
+package schema
+
+import (
+	"strings"
+)
+
+const (
+	// thingfulSchema is the base url for Thingful's ontology
+	thingfulSchema = "https://thingful.github.io/schema#"
+
+	// m3liteSchema is the base url for the m3-lite ontology
+	m3liteSchema = "http://purl.org/iot/vocab/m3-lite#"
+
+	// ssnSchema is the base url for the SSN ontology
+	ssnSchema = "http://purl.oclc.org/NET/ssnx/ssn#"
+
+	// iotliteSchema is the base url for the IOT-LITE ontoloty
+	iotliteSchema = "http://purl.oclc.org/NET/UNIS/fiware/iot-lite#"
+
+	// xsdSchema is the base url for XSD terms
+	xsdSchema = "http://www.w3.org/2001/XMLSchema#"
+)
+
+var (
+	// expanded is a map containing the schema to compact form mappings.
+	namespaces = map[string]string{
+		"thingful:": thingfulSchema,
+		"m3-lite:":  m3liteSchema,
+		"ssn:":      ssnSchema,
+		"iot-lite:": iotliteSchema,
+		"xsd:":      xsdSchema,
+	}
+)
+
+// Expand returns the expanded version of a value (either prop or value). If we
+// are unable to expand the value then we return the input string. This allows
+// this function to be called safely on an already expanded property.
+func Expand(val string) string {
+	// k is the compact version, v is the expanded, and val is a compact string
+	for k, v := range namespaces {
+		if strings.Contains(val, k) {
+			return strings.Replace(val, k, v, 1)
+		}
+	}
+
```

Comment:
What I was thinking was based on something that @dotmancando proposed when we started working on the semantic service. 
The idea was to allow fetcher developers to reference any existing ontology when describing facts, as long as the ontology can be accessed online and imported into Pomelo.

If the `schema.Expand` method returns the original value when nothing matches any known ontology and a developer uses a term taken from a new vocabulary,  a semantically meaningless node would be generated in our semantic service. There is no certainty that a valid URL will always correspond to an existing ontology but enforcing this rule could prevent the usage of invalid namespaces.

Please ignore my original comment if the only allowed ontologies - at least for the near future - are the ones defined in the schema package and what I have proposed is a pre-optimisation not required at this stage (which may well be the case). 

I don't think any work has been done yet to detect and import new ontologies but I quite like the idea of not limiting pomelos semantic knowledge to a strict set of vocabularies.